terraform/examples/offline.tf

```diff
@@ -0,0 +1,761 @@
+# This file is meant to define the 'offline' environment, which lives in the 'crash' trust zone.
+# See also https://github.com/zinfra/backend-issues/wiki/trust-zone-environments (TODO: move that page over to either the zinfra/backend-wiki's wiki, or to zinfra/backend-wiki's repository)
+
+# What is an Offline VPC?
+# The only host reachable from the outside is a bastion host. It has internet access, and provides SSH services. It can access all of the other hosts in the VPC via SSH only.
+
+# To deploy this file, you will need a user with the policies "AmazonEC2FullAccess", "IAMFullAccess", "AmazonS3FullAccess", and "AmazonDynamoDBFullAccess".
+# FUTUREWORK: drill down on the above.
+
+terraform {
+  required_version = ">= 0.12.0"
+
+  backend "s3" {
+    encrypt = true
+    region  = "eu-central-1"
+
+    # TODO: create IAM policy which only allows access to this bucket under
+    # envrionments/crash and to the -crash dynamodb table
+    bucket = "z-terraform-remote-state"
+
+    key = "environments/offline/bootstrap"
+
+    dynamodb_table = "z-terraform-state-lock-dynamo-lock-environment-offline"
+  }
+}
+
+# In AWS, (eu-central-1)
+provider "aws" {
+  region = "eu-central-1"
+}
+
+# Used for the in-VPC EC2 endpoint.
+data "aws_security_group" "default" {
+  name   = "default"
+  vpc_id = module.vpc.vpc_id
+}
+
+module "vpc" {
+  source = "github.com/terraform-aws-modules/terraform-aws-vpc"
```

Comment:
Can you version lock this? Example syntax: `source = "git::https://github.com/cloudposse/terraform-aws-s3-website.git?ref=tags/0.8.0"`